joomla/joomla-cms
# Use JLanguageHelper::getContentLanguages in languagefilter and mod_languages

### Summary of Changes

Use the new JLanguageHelper::getContentLanguages() in language filter plugin and mod_languages.

also:
- add a fallback in JLanguageHelper::getContentLanguages() to en-GB language when no content language is published to avoid redirect loops when there is no content language. - This corrects a bug in language filter when all content languages are unpublished ... redirect loop.
- moved content languages metadesc and keywords from site app to language filter after dispatch event (check if there are no issues with this)

### Testing Instructions

Code review (https://github.com/joomla/joomla-cms/pull/13038/files?w=1 to review without whitespace changes)

1. Use multilingual site and apply patch
2. Confirm all language stuff works properly (add/remove content languages, change order, edit, etc and check the language filter and mod_languages work properly).
3. Confirm content language sitename, description and metakeys works fine.

### Documentation Changes Required

None.

pinging @alikon and @infograf768
